apps/shared/interval_parameter_controller.cpp

```diff
@@ -16,6 +17,11 @@ IntervalParameterController::IntervalParameterController(Responder * parentRespo
   }
 }
 
+void IntervalParameterController::setInterval(Interval * interval) { 
+  m_interval = interval;
+  m_tempInterval = *interval;
```

Comment:
There are heavy members in m_tempInterval that we do not need to keep track of ( m_intervalBuffer ).
It would be lighter to only keep track of m_start, m_end and m_step.
It could be achieved by storing them in a nested class in Interval (such as Interval::IntervalParameters) , and manipulating this class to store the temporary states.